script.rb

```diff
@@ -0,0 +1,144 @@
+# rubocop:disable Style/CaseEquality, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      return to_enum unless block_given?
+
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      return to_enum unless block_given?
+
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      return to_enum unless block_given?
```

Comment:
* [ ] let us improve this line by replacing this ``` return to_enum unless block_given?``` with
``` return to_enum(my_select) unless block_given?```

Reply:
method updated